.github/workflows/main.yml

```diff
@@ -0,0 +1,272 @@
+          [registries.search]
+            registries = ['docker.io']
+          EOF
+
+      - name: install/configure dependencies, privileged Buildah
+        if: ${{ startsWith(matrix.builder, 'buildah-') }}
+        run: |
+          sudo usermod --add-subuids 10000-65536 $USER
+          sudo usermod --add-subgids 10000-65536 $USER
+
+      - name: build/install from Git
+        run: |
+          ./autogen.sh
+          # Remove Autotools to make sure everything works without them.
+          sudo apt-get remove autoconf automake
+          # Configure and verify output.
+          ./configure --prefix=$ch_prefix/from-git
+          set -x
+          fgrep 'documentation: yes' config.log
+          [[ $CH_BUILDER = buildah* ]] && fgrep 'with Buildah: yes' config.log
+          [[ $CH_BUILDER = docker ]] && fgrep 'with Docker: yes' config.log
+          [[ $CH_BUILDER = ch-grow ]] && fgrep 'with ch-grow(1): yes' config.log
+          fgrep 'at least one builder ... yes' config.log
+          fgrep 'ch-run(1): yes' config.log
+          fgrep 'complete test suite: yes' config.log
+          set +x
+          # Build and install.
+          make $ch_makej
+          sudo make $ch_makej install
+          bin/ch-run --version
+          $ch_prefix/from-git/bin/ch-run --version
+
+      - name: late setup & validation
+        run: |
+          bin/ch-test --is-pedantic all
+          bin/ch-test --is-sudo all
+
+      - name: make filesystem permissions fixtures
+        run: |
+          bin/ch-test mk-perm-dirs
+
+      - name: configure sudo to user root, group non-root
+        if: ${{ matrix.keep_sudo }}
+        run: |
+          sudo sed -Ei 's/=\(ALL\)/=(ALL:ALL)/g' /etc/sudoers.d/runner
+          sudo cat /etc/sudoers.d/runner
+
+      - name: remove sudo
+        if: ${{ ! matrix.keep_sudo }}
+        run: |
+          sudo rm /etc/sudoers.d/runner
+          ! sudo echo hello
+
+      - name: build/install from tarball
+        if: ${{ matrix.builder == 'docker' && matrix.keep_sudo }}
+        run: |
+          # Create and unpack tarball.
+          make $ch_makej dist
+          mv charliecloud-*.tar.gz $ch_prefix
+          cd $ch_prefix
+          tar xf charliecloud-*.tar.gz
+          rm charliecloud-*.tar.gz  # else cd fails with "too many arguments"
+          cd charliecloud-*
+          pwd
+          # Configure and verify output.
+          ./configure --prefix=$ch_prefix/from-tarball
+          set -x
+          fgrep 'documentation: yes' config.log
+          [[ $CH_BUILDER = buildah* ]] && fgrep 'with Buildah: yes' config.log
+          [[ $CH_BUILDER = docker ]] && fgrep 'with Docker: yes' config.log
+          [[ $CH_BUILDER = ch-grow ]] && fgrep 'with ch-grow(1): yes' config.log
+          fgrep 'at least one builder ... yes' config.log
+          fgrep 'ch-run(1): yes' config.log
+          fgrep 'complete test suite: yes' config.log
+          set +x
+          # Build and install.
+          make $ch_makej
```

Comment:
Are you suggesting `make "$ch_makej"`?

We control `$ch_makej`, so we know there aren't any quoting related gotchas. If it's empty, it also goes away instead of putting an empty arg in the command (which would confuse `make`); that said we don't have any plans to make it empty. Also the quotes are so ugly. I guess if we started ShellChecking this YAML (not sure how to do that) I'd rather put an exclude rather than quoting it. Opinions?

Reply:
> Are you suggesting `make "$ch_makej"`?

Yes.

>  I guess if we started ShellChecking this YAML (not sure how to do that) I'd rather put an exclude rather than quoting it. Opinions?

I wasn't suggesting we Shellcheck the YAML file itself (I too don't know how we would do that and don't think it would be helpful). I was thinking that the shell commands inside the YAML file end up being converted to a shell script, and I wondered what Shellcheck would have to say about that script.

In anycase, just a thought. I'm not convinced it is an issue.

Reply:
OK. I'm glad you mentioned it though!